src/vs/editor/contrib/wordHighlighter/browser/wordHighlighter.ts

```diff
@@ -675,10 +677,35 @@ class WordHighlighter {
 			// 		1) we have text focus, and a valid query was updated.
 			// 		2) we do not have text focus, and a valid query is cached.
 			// the query will ALWAYS have the correct data for the current highlight request, so it can always be passed to the workerRequest safely
-			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo || WordHighlighter.query.modelInfo.model.isDisposed()) {
+			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo) {
 				return;
 			}
-			this.workerRequest = this.computeWithModel(WordHighlighter.query.modelInfo.model, WordHighlighter.query.modelInfo.selection, WordHighlighter.query.word, otherModelsToHighlight);
+			const queryModelRef = await this.textModelService.createModelReference(WordHighlighter.query.modelInfo.modelURI);
+			const queryModel = queryModelRef.object.textEditorModel;
+			this.workerRequest = this.computeWithModel(queryModel, WordHighlighter.query.modelInfo.selection, otherModelsToHighlight);
+
+			this.workerRequest?.result.then(data => {
+				if (myRequestId === this.workerRequestTokenId) {
+					this.workerRequestCompleted = true;
+					this.workerRequestValue = data || [];
+					this._beginRenderDecorations();
+				}
+			}, onUnexpectedError);
+		} else if (this.model.uri.scheme === Schemas.vscodeNotebookCell) {
+			// new wordHighlighter coming from a different model, NOT the query model, need to create a textModel ref
+
+			// this._stopAll(multiFileConfigChange ? this.model.uri : undefined);
+
+			const myRequestId = ++this.workerRequestTokenId;
+			this.workerRequestCompleted = false;
+
+			if (!WordHighlighter.query || !WordHighlighter.query.modelInfo) {
+				return;
+			}
+
+			const queryModelRef = await this.textModelService.createModelReference(WordHighlighter.query.modelInfo.modelURI);
```

Comment:
Why is this needed and where is this reference disposed?

Reply:
same as above comment.